python/src/aac/plugins/generate/generate_impl.py

```diff
@@ -302,12 +332,47 @@ def generate(
     # build out properties
     # the templates need data from the plugin model to generate code
     context = LanguageContext()
-    parsed_definitions = parse(
-        aac_file
-    )  # we only want to parse, not load, to avoid chicken and egg issues
 
-    # TO DO Consider surrounding this call to parse_and_load with a try except block to catch and handle the various exceptions that can be thrown
-    generator_definitions = context.parse_and_load(generator_file)
+    try:
+        parsed_definitions = parse(
+            aac_file
+        )  # we only want to parse, not load, to avoid chicken and egg issues
+        generator_definitions = context.parse_and_load(generator_file)  # Now we want to parse and load for real
+    except LanguageError as le:
+        status = ExecutionStatus.CONSTRAINT_FAILURE
+        messages.append(
+            ExecutionMessage(
+                message="LanguageError from parse_and_load: " + le.message + " at location: " + le.location,
+                level=MessageLevel.DEBUG,
+                source=aac_file,
+                location=None,  # Included in the message above. Their type/format is not easily compatible with the SourceLocation needed here.
+            )
+        )
+        return (None, ExecutionResult(plugin_name, "check", status, messages))
+    except ParserError as pe:
+        status = ExecutionStatus.PARSER_FAILURE
+
+        messages.append(
+            ExecutionMessage(
+                message="ParserError from parse_and_load. " + process_parser_error(pe),
+                level=MessageLevel.DEBUG,
+                source=aac_file,
+                location=None,  # Included in the message above. Their type/format is not easily compatible with the SourceLocation needed here.
+            )
+        )
+        return (None, ExecutionResult(plugin_name, "check", status, messages))
+    except Exception as ex:  # Also catching IOError but we don't do anything special handling for IOError so handle it here
+        status = ExecutionStatus.GENERAL_FAILURE
+
+        messages.append(
+            ExecutionMessage(
+                message="Exception from parse_and_load. " + ex.message,
+                level=MessageLevel.DEBUG,
+                source=aac_file,
+                location=None,  # Included in the message above. Their type/format is not easily compatible with the SourceLocation needed here.
+            )
+        )
+        return (None, ExecutionResult(plugin_name, "check", status, messages))
```

Comment:
Pretty sure you just copied too far, don't think this line should be here.

Reply:
Nvm I see what this is coming from, viewing in GitPod made it clearer. But, needs tweaked, is still a copy pasta issue
```suggestion
        return (None, ExecutionResult(plugin_name, "generate", status, messages))
```

Reply:
Which line? The return?